common/page.go

```diff
@@ -990,18 +1110,21 @@ func (p *Page) NavigationTimeout() time.Duration {
 // On subscribes to a page event for which the given handler will be executed
 // passing in the ConsoleMessage associated with the event.
 // The only accepted event value is 'console'.
-func (p *Page) On(event string, handler func(*ConsoleMessage)) error {
-	if event != eventPageConsoleAPICalled {
-		return fmt.Errorf("unknown page event: %q, must be %q", event, eventPageConsoleAPICalled)
+func (p *Page) On(event string, handler func(any)) error {
+	switch event {
+	case EventPageConsoleAPICalled:
+	case EventPageMetricCalled:
+	default:
+		return fmt.Errorf("unknown page event: %q", event)
 	}
```

Comment:
Generics can also prevent checking the event type. As far as I can tell, we did this check twice, which can be error-prone when we extend the events.

Reply:
I've linked this comment to a [comment](https://github.com/grafana/xk6-browser/issues/1227#issuecomment-2393348036) in the generalize `page.on` events [issue](https://github.com/grafana/xk6-browser/issues/1227).

We can tackle the generalization after merging in this PR 👍 

I'll resolve this comment so that I can work on the other feedback.